src/components/meta-data/index.jsx

```diff
@@ -0,0 +1,496 @@
+						// eslint-disable-next-line react/no-array-index-key
+						key={`custom-meta-data-${i}`}
+						name={metaObj.metaName}
+						default={metaObj.metaValue}
+					/>
+				))}
+		</>
+	);
+};
+
+const buildUrl = (domain, path) => {
+	try {
+		const url = new URL(path || "", domain);
+		return formatURL(url.href);
+	} catch {
+		return null;
+	}
+};
+
+const getUrlParameters = (requestUri = "") => {
+	const matches = Array.from(
+		requestUri.matchAll(/(?:[&?]?([\w\d%\-._~]{1:100})=([\w\d%\-._~]{1:100})){:10}?/gi)
+	);
+	return matches.reduce((accumulator, [, key, value]) => {
+		if (accumulator[key]) {
+			return {
+				...accumulator,
+				[key]: [
+					...(typeof accumulator[key] === "object" ? accumulator[key] : [accumulator[key]]),
+					value,
+				],
+			};
+		}
+		return { ...accumulator, [key]: value };
+	}, {});
+};
+
+const getPageCanonicalUrl = (pageType, domain, globalContent, requestUri) => {
+	const urlParameters = getUrlParameters(requestUri);
+	const authorCanonicalPath = globalContent?.authors ? globalContent?.authors[0]?.bio_page : "";
+	const globalCanonicalPath = globalContent?.canonical_url;
+	const querylyCanonicalPath = urlParameters.query
+		? `${requestUri.replace(/\?.*/, "")}?query=${urlParameters.query}`
+		: "";
+	const searchCanonicalPath = globalContent?.metadata ? `search/${globalContent?.metadata.q}/` : "";
+	const tagCanonicalPath = globalContent?.Payload ? `tags/${globalContent?.Payload[0]?.slug}/` : "";
+
+	const canonicalUrlMapping = {
+		article: buildUrl(domain, globalCanonicalPath),
+		author: buildUrl(domain, authorCanonicalPath),
+		gallery: buildUrl(domain, globalCanonicalPath),
+		homepage: domain,
+		"queryly-search": buildUrl(domain, querylyCanonicalPath),
+		search: buildUrl(domain, searchCanonicalPath), // arc search
+		section: buildUrl(domain, globalContent?._id || ""),
+		tag: buildUrl(domain, tagCanonicalPath),
+		video: buildUrl(domain, globalCanonicalPath),
+	};
+	return canonicalUrlMapping[pageType];
+};
+
+const generateUrl = (arcSite, websiteDomain, gc) => {
+	const siteData = gc && gc.websites && gc.websites[arcSite];
+	if (!siteData) {
+		return null;
+	}
+	return `${websiteDomain}${siteData.website_url}`;
+};
+
+const normalizeFallbackImage = (websiteDomain, url) => {
+	if (!url) {
+		return null;
+	}
+	if (!url.startsWith("http")) {
+		const tmp = `${websiteDomain}¬${url}`;
+		return tmp.replace(/\/?¬\/?/, "/");
+	}
+	return url;
+};
+
+const MetaData = ({
+	arcSite,
+	canonicalDomain,
+	canonicalResolver,
+	facebookAdmins,
+	fallbackImage,
+	globalContent: gc,
+	MetaTag,
+	MetaTags,
+	metaValue,
+	outputCanonicalLink,
+	requestUri,
+	resizerURL,
+	twitterUsername,
+	websiteDomain,
+	websiteName,
+}) => {
+	const pageType = metaValue("page-type");
+
+	let storyMetaDataTags = null;
+	let tagMetaDataTags = null;
+	let authorMetaDataTags = null;
+	let searchMetaDataTags = null;
+	let sectionMetaDataTags = null;
+	let homepageMetaDataTags = null;
+	let nativoMetaDataTags = null;
```

Comment:
Unless I am missing something, there can only ever be one of these that is not `null`, so do we need separate variables for them? It could just be a single variable, like `pageMetaDataTags`, and then replace all of the uses of these page type variables with that one.